### PR TITLE
Ported eraTcgtt

### PR DESCRIPTION
**Pull request by prakharcode**
progress for #1 

ported eraTcgtt to pure Julia with test cases!

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -160,6 +161,8 @@ AstronomicalTime.update()
         @test AstronomicalTime.Epochs.ut1tai(julian2(ut1), julian1(ut1), dat_ut1) == ERFA.ut1tai(julian2(ut1), julian1(ut1), dat_ut1)
         @test AstronomicalTime.Epochs.taiut1(julian1(tai), julian2(tai), dat_tai) == ERFA.taiut1(julian1(tai), julian2(tai), dat_tai)
         @test AstronomicalTime.Epochs.taiut1(julian2(tai), julian1(tai), dat_tai) == ERFA.taiut1(julian2(tai), julian1(tai), dat_tai)
+        @test +(AstronomicalTime.Epochs.tcgtt(julian1(tcg), julian2(tcg))...) ≈ +(ERFA.tcgtt(julian1(tcg), julian2(tcg))...)
+        @test +(AstronomicalTime.Epochs.tcgtt(julian2(tcg), julian1(tcg))...) ≈ +(ERFA.tcgtt(julian2(tcg), julian1(tcg))...)
```
Comment (helgee) [inline test/runtests.jl]: A simpler way would be:
```julia
sum(AstronomicalTime.Epochs.tcgtt(julian1(tcg), julian2(tcg))) ≈ sum(ERFA.tcgtt(julian1(tcg), julian2(tcg)))
```

Did you investigate why the stricter test (`==`) is not passing?

File: src/conversions.jl [hunk 2]
```diff
@@ -89,6 +89,34 @@ function taiut1(jd1, jd2, dta)
...
+```jldoctest
+julia> tcg = Epoch{TCG}(2.4578265e6, 0.30440190993249416)
+2017-03-14T07:18:20.325 TCG
+julia> AstronomicalTime.Epochs.tcgtt(tcg.jd1, tcg.jd2)
+(2.4578265e6, 0.3027549702857392)
+
+```
+"""
+function tcgtt(jd1, jd2)
+    t77t = MOD_JD_77 + OFFSET_TT_TAI/ SECONDS_PER_DAY
```
Comment (helgee) [inline src/conversions.jl]: Whitespace around `/`

File: test/runtests.jl (hunk 1 above)
Comment (prakharcode) [reply test/runtests.jl]: Yes, it's due to the error in floating point precision.

Comment (prakharcode) [reply test/runtests.jl]: tcg = Epoch{TCG}(2.4578265e6, 0.30440190993249416)
>2017-03-14T07:18:20.325 TCG

Comment (helgee) [reply test/runtests.jl]: Can you give me an example?

Comment (prakharcode) [reply test/runtests.jl]: julia> ERFA.tcgtt(julian1(tcg), julian2(tcg))
(2.4578265e6, 0.3043916774088325)

julia> AstronomicalTime.Epochs.tcgtt(julian1(tcg), julian2(tcg))
(2.4578265e6, 0.3027549702857392)

julia> tcg = Epoch{TCG}(2.4578265e6, 0.30440190993249416)
2017-03-14T07:18:20.325 TCG

julia> JD = ERFA.tcgtt(julian1(tcg), julian2(tcg))
(2.4578265e6, 0.3043916774088325)

julia> TCGEpoch(JD...)
2017-03-14T07:18:19.441 TCG

**Comment by helgee** [comment]
Numerical discrepancies are not the answer here, they only occur when calls to libm are involved (e.g. `sin`, `exp`, etc.). When we are dealing with simple arithmetic the results should be identical to the C version.

I know what's wrong with your code 😉 but I think you should try to figure it out yourself.

File: src/constants.jl [hunk 3]
```diff
@@ -33,5 +33,5 @@ const YEARS_PER_CENTURY    = 100.0
 const OFFSET_TT_TAI = 32.184
 
 const MOD_JD_77 = 43144.0
-const MOD_JD00 = 51544.5
+const MOD_JD0 = 2400000.5
```
Comment (helgee) [inline src/constants.jl]: The constant already exists 😉

Comment (helgee) [reply src/constants.jl]: See above...

Comment (prakharcode) [reply src/constants.jl]: Done :1234:

**Pull request merged**